Libraries/LibGfx/JPGLoader.cpp

```diff
@@ -576,6 +599,31 @@ static bool read_reset_marker(BufferStream& stream, JPGLoadingContext& context)
     return true;
 }
 
+static bool huffman_table_reset_helper(HuffmanTableSpec& src, JPGLoadingContext& context)
+{
+    auto& table = src.type == 0 ? context.dc_tables : context.ac_tables;
+    if (src.destination_id == 0) {
+        if (table.is_empty())
+            table.append(move(src));
+        else
+            table[0] = move(src);
+    }
+
+    if (src.destination_id == 1) {
+        if (table.size() < 1) {
+            String table_str = src.type == 0 ? "DC" : "AC";
+            dbg() << String::format("%s[1] showed up before %s[0]!", table_str, table_str);
```

Comment:
You cannot pass a `String` directly to `String::format()` like this, you need to pass the `String::characters()` (same as `printf`).
Also, let's not create unnecessary temporary strings with `String::format()` when we can just stream to `dbg()`:

```suggestion
            dbg() << table_str << "[1] showed up before " << table_str << "[0]!";
```